protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -119,7 +121,15 @@ public T parse(InputStream stream) {
         }
         CodedInputStream cis = null;
         try {
-          if (stream instanceof KnownLength) {
+          if (stream instanceof NioBufferBackedStream) {
+            int size = stream.available();
+            if (size > 0) {
+              NioBufferBackedStream nioBufferBackedStream = (NioBufferBackedStream) stream;
+              cis = CodedInputStream.newInstance(nioBufferBackedStream.nioBuffer());
```

Comment:
Maybe we can use `ByteBuffer.duplicate()` or `slice()` to create a new ByteBuffer that shares the same content. Because it only reads from the byte buffer here so it's safe to duplicate.